src/lib/Libutil/misc_utils.c

```diff
@@ -718,60 +718,65 @@ replace(char *str, char	*sub, char *repl, char	*retstr)
 
 /**
  * @brief
- *	Add esc as escape character if delim is appearing on str
+ *	Escape every occurrence of 'delim' in 'str' with 'esc'
  *
- * @param[in]	str		- input string
- * @param[in]	delim  	- delimiter to be searched on str
- * @param[in]   esc 	- escape character to be added if delim found on str
- * @param[out]  retstr 	- delim replaced with and escape character+delim if found.
+ * @param[in]	str     - input string
+ * @param[in]	delim   - delimiter to be searched in str
+ * @param[in]	esc     - escape character to be added if delim found in str
  *
- * @return void
+ * @return	string
+ * @retval	NULL	- memory allocation failed or str is NULL
+ * @retval	retstr	- output string, with every occurrence of delim escaped with 'esc'
  */
 
-void
-escape_delimiter(char *str, char *delim, char esc, char *retstr)
+char *
+escape_delimiter(char *str, char *delim, char esc)
 {
 	int	i=0, j=0;
-	int	delim_len;
-	char	rstr[MAXBUFLEN+1];
+	int	delim_len=0;
+	int retstrlen=0;
+	char *retstr = NULL;
+	char *temp = NULL;
 
 	if (str == NULL)
-		return;
+		return NULL;
 
-	if (*str == '\0') {
-		retstr[0] = '\0';
-		return;
-	}
-
-	if ((delim == NULL || *delim == '\0') || esc == '\0') {
-		strncpy(retstr, str, strlen(str));
-		return;
+	if (*str == '\0' || (delim == NULL || *delim == '\0') || esc == '\0') {
+		return strdup((char *)str);
 	}
 	delim_len = strlen(delim);
+	retstr = (char *)malloc(MAXPATHLEN);
+	if(retstr == NULL)
+		return NULL;
+	retstrlen = MAXPATHLEN;
 
 	while (*str != '\0') {
-		if( *str == esc && i < MAXBUFLEN) {
-				rstr[i++] = esc;
-				rstr[i++] = esc;
+		if(*str == esc) {
+			retstr[i++] = esc;
+			retstr[i++] = esc;
 				str++;
-		} else if(strncmp(str, delim, delim_len) == 0 && i <= MAXBUFLEN) {
-				rstr[i++]=esc;
-			for (j=0; (j < delim_len && i <= MAXBUFLEN); j++, i++) {
-				rstr[i] = delim[j];
+		} else if(strncmp(str, delim, delim_len) == 0) {
```

Comment:
@nithinj is there a need in pbsnodes to escape the delimiter sequence? If so, how do we do that in pbsnodes? My guess is that there is no such need because pbsnodes does not deal with Variable_List like qstat does.
Would you agree that doing a strncmp after copying every character could have serious implications on performance of this command? 
I'd suggest that you please check that if there is no use case to have more than a single character delimiter then we can drop this requirement.